test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -49,164 +54,234 @@ class database_operation {
+     * Basic update operation that uses a random cursor to update values in a randomly chosen
+     * collection.
      */
     virtual void
     update_operation(thread_context *tc)
     {
+        /* A structure that's used to track which cursors we've opened for which collection. */
+        struct collection_cursors {
+            const std::string collection_name;
+            WT_CURSOR *random_cursor;
+            WT_CURSOR *update_cursor;
+        };
+        debug_print(
+          type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing loop.",
+          DEBUG_INFO);
         WT_CURSOR *cursor;
         WT_DECL_RET;
         wt_timestamp_t ts;
-        std::vector<WT_CURSOR *> cursors;
-        std::vector<std::string> collection_names = tc->database.get_collection_names();
+        std::map<uint64_t, collection_cursors> collections;
         key_value_t key, generated_value;
+        std::string collection_name;
         const char *key_tmp;
-        uint64_t i = 0;
+        uint64_t collection_count = 0, collection_id = 0;
         bool using_timestamps = tc->timestamp_manager->enabled();
 
-        /* Get a cursor for each collection in collection_names. */
-        for (const auto &it : collection_names) {
-            testutil_check(tc->session->open_cursor(tc->session, it.c_str(), NULL, NULL, &cursor));
-            cursors.push_back(cursor);
-        }
-
         /*
-         * Update each collection while the test is running.
+         * Loop while the test is running.
          */
-        while (tc->running() && !collection_names.empty()) {
-            if (i >= collection_names.size())
-                i = 0;
-            ret = cursors[i]->next(cursors[i]);
-            /* If we have reached the end of the collection, reset. */
-            if (ret == WT_NOTFOUND) {
-                testutil_check(cursors[i]->reset(cursors[i]));
-                ++i;
-            } else if (ret != 0)
-                /* Stop updating in case of an error. */
-                testutil_die(DEBUG_ERROR, "update_operation: cursor->next() failed: %d", ret);
-            else {
-                testutil_check(cursors[i]->get_key(cursors[i], &key_tmp));
+        while (tc->running()) {
+            /*
+             * Sleep the period defined by the op_rate in the configuration. Do this at the start of
+             * the loop as it could be skipped by a subsequent continue call.
+             */
+            tc->sleep();
+
+            /* Pick a random collection to update if there are any, taking care to subtract -1. */
+            collection_count = tc->database.get_collection_count();
+            if (collection_count == 0)
+                continue;
+
+            collection_id =
+              random_generator::instance().generate_integer<uint64_t>(0, collection_count - 1);
+            if (collections.find(collection_id) == collections.end()) {
```

Comment:
`find` gives you an iterator so there's no need to do two lookups here.
No need to block the merge for this though.

Reply:
Will leave as is.